src/Stryker.Core/Stryker.Core/Reporters/Html/HtmlReporter.cs

```diff
@@ -0,0 +1,148 @@
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+using Spectre.Console;
+using Stryker.Core.Mutants;
+using Stryker.Core.Options;
+using Stryker.Core.Options.Inputs;
+using Stryker.Core.ProjectComponents;
+using Stryker.Core.Reporters.Html.ProcessWrapper;
+using Stryker.Core.Reporters.Html.Realtime;
+using Stryker.Core.ProjectComponents.TestProjects;
+using Stryker.Core.Reporters.Json;
+
+namespace Stryker.Core.Reporters.Html;
+
+public class HtmlReporter : IReporter
+{
+    private readonly StrykerOptions _options;
+    private readonly IFileSystem _fileSystem;
+    private readonly IAnsiConsole _console;
+    private readonly IWebbrowserOpener _processWrapper;
+    private readonly IRealtimeMutantHandler _mutantHandler;
+
+    public HtmlReporter(
+        StrykerOptions options,
+        IFileSystem fileSystem = null,
+        IAnsiConsole console = null,
+        IWebbrowserOpener processWrapper = null,
+        IRealtimeMutantHandler mutantHandler = null)
+    {
+        _options = options;
+        _fileSystem = fileSystem ?? new FileSystem();
+        _console = console ?? AnsiConsole.Console;
+        _processWrapper = processWrapper ?? new WebbrowserOpener();
+        _mutantHandler = mutantHandler ?? new RealtimeMutantHandler(_options);
+    }
+
+    public void OnAllMutantsTested(IReadOnlyProjectComponent reportComponent, TestProjectsInfo testProjectsInfo)
+    {
+        var reportPath = BuildReportPath();
+        WriteHtmlReport(reportPath, reportComponent, testProjectsInfo);
+
+        if (_options.ReportTypeToOpen == ReportType.Html)
+        {
+            _mutantHandler.CloseSseEndpoint();
+            return;
+        }
+        var reportUri = "file://" + reportPath.Replace("\\", "/").Replace(" ", "%20");
+
+        var green = new Style(Color.Green);
+        _console.WriteLine();
+        _console.WriteLine("Your html report has been generated at:", green);
+
+        if (_console.Profile.Capabilities.Links)
+        {
+            // We must print the report path as the link text because on some terminals links might be supported but not actually clickable: https://github.com/spectreconsole/spectre.console/issues/764
+            _console.WriteLine(reportPath, green.Combine(new Style(link: reportUri)));
+        }
+        else
+        {
+            _console.WriteLine(reportUri, green);
+        }
+
+        _console.WriteLine("You can open it in your browser of choice.", green);
+    }
+
+    private void OpenReportInBrowser(IReadOnlyProjectComponent reportComponent, string reportPath, TestProjectsInfo testProjectsInfo)
```

Comment:
private methods should not be in-between public methods, move them to below the last public method